src/value.c

```diff
@@ -275,9 +351,58 @@ SIValue SIValue_Divide(const SIValue a, const SIValue b) {
 	return SI_DoubleVal(SI_GET_NUMERIC(a) / (double)SI_GET_NUMERIC(b));
 }
 
+int SIArray_Compare(SIValue arrayA, SIValue arrayB) {
+	uint arrayALen = SIArray_Length(arrayA);
+	uint arrayBLen = SIArray_Length(arrayB);
+	// check empty list
+	if(arrayALen == 0 && arrayBLen == 0) return 0;
+	int lenDiff = arrayALen - arrayBLen;
+	// check for the common range of indices
+	uint minLengh = arrayALen < arrayBLen ? arrayALen : arrayBLen;
+
+	uint allDisjoint = 0;   // counter for the amount of disjoint comparisons
+	uint nullCompare = 0;   // counter for the amount of null comparison
+	int notEqual =
+		0;       // will hold the first false (result != 0) comparison result between two values from the same type, which are not equal
+
+	// go over the common range for both arrays
+	for(uint i = 0; i < minLengh; i++) {
+		SIValue aValue = SIArray_Get(arrayA, i);
+		SIValue bValue = SIArray_Get(arrayB, i);
+		int compareResult = SIValue_Compare(aValue, bValue);
+		switch(compareResult) {
+		case 0:
+			// we have a matching value
+			break;
+		case COMPARED_NULL:
+			// there was a null comparison
+			nullCompare++;  // update null comparison counter
+			allDisjoint++;  // null comparison is also a disjoint comparison
+			break;
+		case DISJOINT:
+			allDisjoint++;  // there was a disjoint comparison
+			if(notEqual == 0) notEqual =
+					compareResult; // if there wasn't any false comparison, update the false comparison to disjoint value
+			break;
+		default:
+			// if comparison is not 0 (a != b), set the first value
+			if(notEqual == 0) notEqual = compareResult;
+			break;
+		}
+	}
+	// if all the elements in the shared range are from disjoint types return DISJOINT array
+	if(allDisjoint == minLengh && allDisjoint > nullCompare) return DISJOINT;
+	// if there was a null comperison on non disjoint arrays
+	if(nullCompare) return COMPARED_NULL;
+	// if there was a diffrence in some member
+	if(notEqual) return notEqual;
+	// if all elemnts are equal and length are equal so arrays are equal, otherwise return lengh diff
+	return lenDiff;
+}
```

Comment:
I know it's in the Language Reference v9, but is Cypher still supporting list comparisons like this? I get a syntax error in Neo on `RETURN [1,2] < [1,2]` and there is no mention of lists in https://neo4j.com/docs/cypher-manual/current/syntax/operators/#cypher-ordering .

Reply:
try `unwind [[1,2],[2,3]] as x return x order by x DESC`

also, section 3.2.3 at  https://github.com/opencypher/openCypher/blob/master/cip/1.accepted/CIP2016-06-14-Define-comparability-and-equality-as-well-as-orderability-and-equivalence.adoc